src/Microsoft.Extensions.Options.ConfigurationExtensions/OptionsBuilderConfigurationExtensions.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for adding configuration related options services to the DI container via <see cref="OptionsBuilder{TOptions}"/>.
+    /// </summary>
+    public static class OptionsBuilderConfigurationExtensions
+    {
+        /// <summary>
+        /// Registers a configuration instance which TOptions will bind against.
+        /// </summary>
+        /// <typeparam name="TOptions">The options type to be configured.</typeparam>
+        /// <param name="optionsBuilder">The options builder to add the services to.</param>
+        /// <param name="config">The configuration being bound.</param>
+        /// <returns>The <see cref="OptionsBuilder{TOptions}"/> so that additional calls can be chained.</returns>
+        public static OptionsBuilder<TOptions> Configure<TOptions>(this OptionsBuilder<TOptions> optionsBuilder, IConfiguration config) where TOptions : class
```

Comment:
Actually now that we have a Builder to hang this off of, we don't have to call it Configure, we should call this method `Bind` to give it more meaning

Reply:
I would gladly do that change, but let's clarify the bigger picture first. Does this also apply to the overload that takes and `Action<TOptions>`. And would we rename `PostConfigure` to `PostBind`?
Or maybe those overloads should be renamed to `Apply` and `PostApply`?  Or `Update`?

Reply:
Yeah we probably should switch to something different, its going to be too insane to keep adding all these overloads as longer term I wouldn't be surprised if there were more lifecycles either, like Validation.

Which probably would mean something closer to:

Bind(IConfiguration, OptionsLifecycle.[Configure/PostConfigure/Validation]).

To level we'd still have
Configure
PostConfigure
(Validate in the future)

Reply:
It looks like we don't have PostConfigure(IConfiguration) today anyways, so I don't think you need to worry about it for the purposes of this PR...  We can tackle this in a different issue/PR

Reply:
Its just sugar for `AddOptions<T>.PostConfigure(o => config.Bind(o))` anyways.